src/main/java/core/basesyntax/service/RegistrationServiceImp.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+
+public class RegistrationServiceImp {
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImp(StorageDaoImpl storage) {
```

Comment:
The constructor parameter type should be the interface (StorageDao) instead of the implementation (StorageDaoImpl). This is because we want to program to an interface, not an implementation. This way, if we decide to change the implementation of StorageDao in the future, we will not have to change this constructor.